### PR TITLE
Added week parameter to functions, adjusted final_score to look at previous week

### PR DESCRIPTION
**Pull request by ZachMyers3**
For #55 

The Tuesday scheduled task was running with an incorrect week variable and getting the scores for next week. I've refactored all the functions to take an optional week variable, and default to the current week from the league if none is passed. I've also adjusted the Tuesday final score function to pull the previous week.

### CODE REVIEW

**Comment by cwendt94** [comment]
You can pass None to box_scores() and the API will automatically use the current week so I don’t think you need those extra checks at the top of each function.

**Comment by ZachMyers3** [comment]
Just tried this, power_rankings requires an integer type. I can remove all the others except that, however.

File: ff_bot/ff_bot.py [hunk 1]
```diff
@@ -299,7 +305,9 @@ def bot_main(function):
     elif function=="get_trophies":
         text = get_trophies(league)
     elif function=="get_final":
-        text = "Final " + get_scoreboard_short(league)
+        # on Tuesday we need to get the scores of last week
+        week = league.current_week - 1
+        text = "Final " + get_scoreboard_short(league, week=week)
         text = text + "\n\n" + get_trophies(league)
```
Comment (bberk) [inline ff_bot/ff_bot.py]: Would the get_trophies also need to be `get_trophies(league, week=week)` since the league's current_week is the upcoming week?

**Comment by cwendt94** [comment]
True, power_rankings should always have a week input. The -1 for week will use the current week in the API

File: ff_bot/ff_bot.py (hunk 1 above)
Comment (ZachMyers3) [reply ff_bot/ff_bot.py]: It wouldn't, all of the trophies are calculated off of the box score, which can accept a None type for a week.

**Comment by scuba-fan** [comment]
Waiting for the merge.  Thanks ZachMyers3.

**Comment by ZachMyers3** [comment]
Any reason this hasn't been merged? I'm willing to make changes if need be.

**Pull request merged**